risc0/zkvm/src/guest/env.rs

```diff
@@ -185,27 +184,31 @@ pub fn commit_slice<T: Pod>(slice: &[T]) {
 /// Return the number of processor cycles that have occured since the guest
 /// began.
 pub fn get_cycle_count() -> usize {
-    unsafe { sys_cycle_count() }
+    unsafe {
+        let (a0, _) = syscall!(SYS_CYCLE_COUNT, core::ptr::null_mut(), 0);
```

Comment:
It seems like some syscalls (like `sys_cycle_count`) isn't verify similar to I/O. There might be other syscalls we want to add that also don't really fit that mold. It feels like we're trying to shoehorn all things into I/O, but I'm not sure that makes the most conceptual sense.